host/tests/dsDisplay_L3_Tests/dsDisplay_test01_GetDisplayEDID.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import os
+import sys
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(dir_path+"/../")
+
+from dsClasses.dsDisplay import dsDisplayClass
+from raft.framework.plugins.ut_raft import utHelperClass
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utUserResponse import utUserResponse
+
+class dsDisplay_test01_GetDisplayEDID(utHelperClass):
+
+    """
+    Test class to retrieve and verify the EDID (Extended Display Identification Data) from a display.
+
+    This class uses the `dsDisplayClass` to interact with the device's display,
+    downloading necessary test assets, retrieving the EDID, and performing verification.
+    """
+
+    # Class variables
+    testName  = "test01_GetDisplayEDID"
+    testSetupPath = dir_path + "/dsDisplay_L3_testSetup.yml"
+    moduleName = "dsDisplay"
+    rackDevice = "dut"
+
+    def __init__(self):
+        """
+        Initializes the test03_GetDisplayAspectRatio test .
+
+        Args:
+            None.
+        """
+        super().__init__(self.testName, '1')
+
+        # Test Setup configuration file
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Session for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+
+        # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.deviceProfile = dir_path + "/" + self.cpe.get("test").get("profile")
+
+    def testDownloadAssets(self):
```

Comment:
create a dsDisplayHelper class and move all common code to dsDisplayHelper (refer https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/feature/gh233_dsAudio_L3_python_updates/host/tests/L3_TestCases/dsAudio/dsAudioHelperClass.py) 

Follow the same for all similar cases